test/test_sparse.py

```diff
@@ -1798,24 +1798,27 @@ def test_shape(sparse_dims, nnz, with_size):
     def test_sparse_sum(self, device, dtype, coalesced):
 
         def run_tests(S, td=None):
-            D = S.coalesce().to_dense().detach().requires_grad_(True)
+            D = S.to_dense().requires_grad_(True)
             if td is None:
                 S_sum = torch.sparse.sum(S)
                 D_sum = D.sum()
                 self.assertEqual(S_sum.item(), D_sum.item())
 
                 def fn(S):
+                    S = S.sparse_mask(S)
                     return torch.sparse.sum(S)
-                gradcheck(fn, (S,), masked=True)
+                gradcheck(fn, (S.clone().requires_grad_(True),))
             else:
                 S_sum = torch.sparse.sum(S, td)
                 D_sum = D.sum(td)
                 self.assertEqual(S_sum.to_dense() if S_sum.is_sparse else S_sum, D_sum)
 
                 def fn(S):
                     res = torch.sparse.sum(S, td)
-                    return res.to_dense(masked_grad=True)
-                gradcheck(fn, (S,), masked=True)
+                    if res.layout != torch.strided:
+                        res = res.sparse_mask(res)
+                    return res.to_dense(masked_grad=False)
```

Comment:
Once `sparse_mask` supports CSR/CSC, it would make sense to move it inside the `torch.sparse.sum`'s body.